frontend/src/components/UserInfo.js

```diff
@@ -342,4 +353,83 @@ class AccountProfileField extends React.Component {
     }
 }
 
+function DeleteAccountDialog(user) {
+    const [open, setOpen] = React.useState(false);
+    const [message, setMessage] = React.useState('');
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const deleteAccount = () => {
+        if (message !== 'Delete my account') {
+            alert(
+                'Please type "Delete my account" to verify your intention to delete your account. Case sensitive.',
+            );
+            return;
+        }
+        setOpen(false);
+        apiWrapper
+            .executeCommand('ufarewell ' + user.user.id)
+            .then((data) => {
+                window.location.href = 'https://worksheets.codalab.org';
+            })
+            .catch((error) => alert(error));
+    };
+
+    return (
+        <div>
+            <Divider style={{ marginBottom: '30px' }} />
+            <h2 style={{ color: '#bc3638' }}>Delete account</h2>
+            <p>
+                To be safe, you can only delete your account if you do not own any bundles,
+                worksheets, or groups.
```

Comment:
what does owning a group mean? Does this just mean being part of a group?

Reply:
A group only has one owner, and owning a group means that the user creates the group